astropy/time/core.py

```diff
@@ -1779,14 +1832,14 @@ def sidereal_time(self, kind, longitude=None, model=None):
         kind : str
             ``'mean'`` or ``'apparent'``, i.e., accounting for precession
             only, or also for nutation.
-        longitude : `~astropy.units.Quantity`, `str`, or None; optional
-            The longitude on the Earth at which to compute the sidereal time.
-            Can be given as a `~astropy.units.Quantity` with angular units
-            (or an `~astropy.coordinates.Angle` or
-            `~astropy.coordinates.Longitude`), or as a name of an
-            observatory (currently, only ``'greenwich'`` is supported,
-            equivalent to 0 deg).  If `None` (default), the ``lon`` attribute of
-            the Time object is used.
+        longitude : `~astropy.units.Quantity`, `~astropy.coordinates.EarthLocation`, str, or None; optional
+            The longitude on the Earth at which to compute the Earth rotation
+            angle (taken from a location as needed).  If `None` (default), taken
+            from the ``location`` attribute of the Time instance. If the special
+            string  'greenwich' or 'tio', the result will be relative to longitude
+            0 for models before 2000, and relative to the Terrestrial Intermediate
+            Origin (TIO) for later ones (i.e., the output of the relevant ERFA
+            function that calculates greenwich sidereal time).
```

Comment:
I think this should mention that if a plain number is passed, it is assumed to be degrees, which I think is a bit surprising.

As plain numbers are usually interpreted as radians, right?

Reply:
It is really for backward compatibility for `sidereal_time` - if I had written this now I would have insisted on a unit (and plain `Longitude` input does). So, my sense is not to advertise it.

Reply:
Wouldn't we allow to change this as this targets 5.0? It seems like a good change to me to require a unit

Reply:
Not quite sure whether it is worth it, but I guess we could indeed have a `DeprecationWarning` starting in 5.0.

Reply:
I'm a bit confused by this. The documentation used to say:
```
            The longitude on the Earth at which to compute the sidereal time.
            Can be given as a `~astropy.units.Quantity` with angular units
            (or an `~astropy.coordinates.Angle` or
            `~astropy.coordinates.Longitude`)
```
In fact, as we found out, one can also use a string (e.g. `'10.2345d'`). I'm not sure why this was dropped, but the code hasn't changed in that respect. Isn't the `Longitude` constructor intelligent enough to convert all of these to degrees if given another angular unit?

Reply:
btw, @taldcroft thank you for finally responding!

Reply:
Oh, I think I understand. Although this is not mentioned, one can also use a `float` as the argument, in which case it is assumed to be degrees. I agree with @maxnoe on this. We should have mentioned that in the documentation. Too bad he didn't bring this up a bit earlier. In some respect, I don't thing it matters much though. People don't usually specify the longitude of an observatory in radians anyway. And, if they did, they would soon find out that the result would be incorrect.

Reply:
Yes, I guess it should have been "or anything that can initialize a `Longitude`" - the fact that `float` is also allowed is extra (because we give a default unit of `u.deg` in the initializer). The question is whether it is worth doing a quick follow-up PR that deprecates that behaviour (and perhaps clarifies the docstring).

Reply:
I'll leave that entirely up to you. Obviously, `sidereal_time()` has been around for awhile, so any confusion will only affect new users and people who have been using it for years may not appreciate depreciation warnings, but if you feel strongly that a unit should be specified, then go ahead and require one.

Reply:
@mkbrewer I didn't habe a look at this earlier because I only noticed this PR because the corresponding issue Was closed by merging it ;)

Reply:
And I am very happy to have it closed after being open for over a year!